app/controllers/nodes_controller.rb

```diff
@@ -5,6 +5,16 @@
 # NodesController is responsible for everything related to nodes: showing
 # information on nodes, deleting them, etc.
 class NodesController < ApplicationController
+  rescue_from Minion::NonExistingMinion do |e|
+    respond_to do |format|
+      format.html do
+        flash[:error] = e.message
+        redirect_to nodes_path
+      end
+      format.json { render json: e.message, status: :unprocessable_entity }
```

Comment:
I'd say that the status should be 404 instead, since the problem is that the requester referenced a resource that could not be found.